### PR TITLE
add option to set default querystring params (for API key)

### PR DESCRIPTION
**Pull request by rubensayshi**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,7 +1,46 @@
...
 var jsend = require('jsend')
+var querystring = require('querystring')
+
+/**
+ * helper to copy properties
+ *
+ * @param {Object} target
+ * @returns {Object}
+ */
+function extend(target) {
```
Comment (rubensayshi) [inline index.js]: idk if you want to add `lodash` or something similar to do this or use this simple helper?

Comment (dcousens) [inline index.js]: is there any reason to extend a `{}` rather than just use the object they pass in?

**Comment by dcousens** [comment]
Nits aside, I'm for this change

File: index.js [hunk 2]
```diff
@@ -1,7 +1,46 @@
...
+  var qsKeys = Object.keys(params)
+  qsKeys.sort()
+  for (var i in qsKeys) {
+    var qsKey = qsKeys[i]
+    var qsChunk = {}
+    qsChunk[qsKey] = params[qsKey]
+    query.push(querystring.stringify(qsChunk))
+  }
+  return query.join("&")
+}
```
Comment (dcousens) [inline index.js]: Why not just

``` javascript
function qs(argObj) {
  var result = {}
  Object.keys(argObj).sort().forEach(function(key) { result[key] = argObj[key] })

  return querystring(result)
}
```

File: index.js [hunk 3]
```diff
@@ -1,7 +1,46 @@
...
+    for (var prop in source) {
+      target[prop] = source[prop]
+    }
+  })
+  return target
+}
 
-function req (url, body, plural, xhrOptions, callback) {
+/**
+ * helper to make sure the query string is sorted in lexical order
```
Comment (dcousens) [inline index.js]: For what it is worth, is there a reason to sort them?

Comment (rubensayshi) [inline index.js]: yea, some standards say they should be, except noone seems to care.

except when you want to do something like http://tools.ietf.org/html/draft-cavage-http-signatures-03  when the server and the client need to make sure they use the exact same URL when doing the signing.

File: index.js (hunk 1 above)
Comment (rubensayshi) [inline index.js]: that's the main reason, I guess we can do without right now

File: index.js (hunk 2 above)
Comment (rubensayshi) [inline index.js]: ok :D

File: index.js (hunk 3 above)
Comment (dcousens) [inline index.js]: If that signature signing is the case, won't users want more fidelity over the entire process anyway?  And they could easily just pass in a sorted parameters object as such?

**Pull request merged**